examples/light-switch-app/efr32/src/binding-handler.cpp

```diff
@@ -18,51 +18,164 @@
+    {
+    case Clusters::OnOff::Commands::Toggle::Id:
+        Clusters::OnOff::Commands::Toggle::Type toggleCommand;
+        Controller::InvokeCommandRequest(peer_device->GetExchangeManager(), peer_device->GetSecureSession().Value(), binding.remote,
+                                         toggleCommand, onSuccess, onFailure);
+        break;
+
+    case Clusters::OnOff::Commands::On::Id:
+        Clusters::OnOff::Commands::On::Type onCommand;
+        Controller::InvokeCommandRequest(peer_device->GetExchangeManager(), peer_device->GetSecureSession().Value(), binding.remote,
+                                         onCommand, onSuccess, onFailure);
+        break;
+
+    case Clusters::OnOff::Commands::Off::Id:
+        Clusters::OnOff::Commands::Off::Type offCommand;
+        Controller::InvokeCommandRequest(peer_device->GetExchangeManager(), peer_device->GetSecureSession().Value(), binding.remote,
+                                         offCommand, onSuccess, onFailure);
+        break;
+
+    default:
+        ChipLogError(NotSpecified, "Invalid binding command data - commandId is not supported");
+        break;
+    }
+}
+
+void LightSwitchChangedHandler(const EmberBindingTableEntry & binding, chip::DeviceProxy * peer_device, void * context)
+{
+    VerifyOrReturn(context != nullptr, ChipLogError(NotSpecified, "Invalid context for Light switch handler"););
+    BindingCommandData * data = static_cast<BindingCommandData *>(context);
+
     if (binding.type == EMBER_MULTICAST_BINDING)
     {
         ChipLogError(NotSpecified, "Group binding is not supported now");
-        return;
+    }
+    else if (binding.type == EMBER_UNICAST_BINDING && binding.local == 1 &&
+             (!binding.clusterId.HasValue() || binding.clusterId.Value() == data->clusterId))
+    {
+
+        switch (data->clusterId)
+        {
+        case Clusters::OnOff::Id:
+            ProcessOnOffBindingCommand(data->commandId, binding, peer_device);
+            break;
+        default:
+            ChipLogError(NotSpecified, "Invalid binding command data - clusterId is not supported");
+            break;
+        }
     }
 
-    if (binding.type == EMBER_UNICAST_BINDING && binding.local == 1 && binding.clusterId.HasValue() &&
-        binding.clusterId.Value() == Clusters::OnOff::Id)
+    chip::Platform::Delete(data);
+}
+
+#ifdef ENABLE_CHIP_SHELL
+CHIP_ERROR SwitchCommandHandler(int argc, char ** argv)
+{
+    if (argc == 1 && strcmp(argv[0], "on") == 0)
     {
-        Clusters::OnOff::Commands::Toggle::Type toggleCommand;
-        Controller::InvokeCommandRequest(peer_device->GetExchangeManager(), peer_device->GetSecureSession().Value(), binding.remote,
-                                         toggleCommand, onSuccess, onFailure);
+        chip::DeviceLayer::PlatformMgr().ScheduleWork(SwitchOnOffOn, 0);
+    }
+    else if (argc == 1 && strcmp(argv[0], "off") == 0)
+    {
+        chip::DeviceLayer::PlatformMgr().ScheduleWork(SwitchOnOffOff, 0);
```

Comment:
If this whole file uses namespace chip, a bunch of these `chip::` are not needed.

Reply:
done